src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/integration/StaffResourceIntTest.kt

```diff
@@ -0,0 +1,93 @@
+package uk.gov.justice.digital.hmpps.hmppsassessforearlyreleaseapi.integration
+
+import org.assertj.core.api.Assertions
+import org.junit.jupiter.api.AfterAll
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.springframework.core.ParameterizedTypeReference
+import uk.gov.justice.digital.hmpps.hmppsassessforearlyreleaseapi.integration.base.SqsIntegrationTestBase
+import uk.gov.justice.digital.hmpps.hmppsassessforearlyreleaseapi.integration.wiremock.DeliusMockServer
+import uk.gov.justice.digital.hmpps.hmppsassessforearlyreleaseapi.service.probation.Name
+import uk.gov.justice.digital.hmpps.hmppsassessforearlyreleaseapi.service.probation.User
+
+private const val USERNAME = "com-user"
+private const val GET_STAFF_DETAILS_BY_USERNAME_URL = "/staff?username=$USERNAME"
+
+class StaffResourceIntTest : SqsIntegrationTestBase() {
+
+  @Nested
+  inner class GetStaffDetailsByUsername {
+    @Test
+    fun `should return unauthorized if no token`() {
+      webTestClient.get()
+        .uri(GET_STAFF_DETAILS_BY_USERNAME_URL)
+        .exchange()
+        .expectStatus()
+        .isUnauthorized
+    }
+
+    @Test
+    fun `should return forbidden if no role`() {
+      webTestClient.get()
+        .uri(GET_STAFF_DETAILS_BY_USERNAME_URL)
+        .headers(setAuthorisation())
+        .exchange()
+        .expectStatus()
+        .isForbidden
+    }
+
+    @Test
+    fun `should return forbidden if wrong role`() {
+      webTestClient.get()
+        .uri(GET_STAFF_DETAILS_BY_USERNAME_URL)
+        .headers(setAuthorisation(roles = listOf("ROLE_WRONG")))
+        .exchange()
+        .expectStatus()
+        .isForbidden
+    }
+
+    @Test
+    fun `should return staff details with a username`() {
+      deliusMockServer.stubPostStaffDetailsByUsername()
+
+      val comUserDetails = webTestClient.get()
+        .uri(GET_STAFF_DETAILS_BY_USERNAME_URL)
+        .headers(setAuthorisation(roles = listOf("ASSESS_FOR_EARLY_RELEASE_ADMIN")))
+        .exchange()
+        .expectStatus()
+        .isOk
+        .expectBody(object : ParameterizedTypeReference<User>() {})
+        .returnResult().responseBody!!
+
+      Assertions.assertThat(comUserDetails).isEqualTo(comUser)
+    }
+  }
+
```

Comment:
**suggestion:** I think it may be worth adding a test to ensure it handles returning an empty array

Reply:
worth static importing Assertions, but a test to cover the 404 would be good.

Reply:
done.